### PR TITLE
Increasing Giant Bomb API accuracy

### PR DESCRIPTION
**Pull request by arpanlaha**

### CODE REVIEW

File: backend/api/views/games.py [hunk 1]
```diff
@@ -246,18 +261,69 @@ def get_giantbomb_data(game_name, game_system):
...
+        and gb_dict["image"] == ""
+        and gb_dict["thumbnail"] == ""
+        and modified_name != ""
+    ):
+        # remove last word and retry
+        split_modified = modified_name.split(" ")
+        remove_last = ""
+        for i in range(len(split_modified) - 1):
+            remove_last = remove_last + split_modified[i] + " "
+        modified_name = remove_last.strip()
```
Comment (42B) [inline backend/api/views/games.py]:
```python
modified_name = ' '.join(split_modified[:-1])  # ?
```

File: backend/api/views/games.py [hunk 2]
```diff
@@ -246,18 +261,66 @@ def get_giantbomb_data(game_name, game_system):
...
-                gb_dict["thumbnail"] = result["image"]["icon_url"]
-            if result["image"]["small_url"] is not None:
-                gb_dict["image"] = result["image"]["small_url"]
-            return gb_dict
+    gb_data = requests.get(url=gb_url, params=gb_params, headers=headers).json()
+
+    if len(gb_data["results"]) != 0:
+        best_match = gb_data["results"][0]
+        best_similarity = 0
+        logger.info("initialized")
```
Comment (HRimaw2) [inline backend/api/views/games.py]: logger lines

File: backend/api/views/games.py [hunk 3]
```diff
@@ -246,18 +261,65 @@ def get_giantbomb_data(game_name, game_system):
...
+        if best_match["image"]["small_url"] is not None:
+            gb_dict["image"] = best_match["image"]["small_url"]
+
+    while (
+        gb_dict["description"] == ""
+        and gb_dict["image"] == ""
+        and gb_dict["thumbnail"] == ""
+        and modified_name != ""
+    ):
+        # remove last word and retry
```
Comment (abetco) [inline backend/api/views/games.py]: Maybe just have a if statement that removes the last word if it's not the first time through the while loop?
Just to try and remove the repetitive code above the while loop.
Something like
boolean remove_end = false
while {
      if (remove_end) : 
             remove the last word
      get results from api
      look for closest match
      remove_end = true
}

**Review by abetco — APPROVED** [review]
:)

**Pull request merged**